hadoop-hdds/test-utils/src/main/java/org/apache/ozone/test/tag/Flaky.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.ozone.test.tag;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.junit.jupiter.api.Tag;
+
+/**
+ * Annotation to mark test classes or methods with some intermittent failures.
+ * These are handled separately from the normal tests.  (Not required to pass,
+ * may be repeated automatically, etc.)
+ */
+@Target({ ElementType.TYPE, ElementType.METHOD })
+@Retention(RetentionPolicy.RUNTIME)
+@Tag("flaky")
+public @interface Flaky {
```

Comment:
Looks like the implementation is a little different from what we discussed. I thought we wanted to created 3 or 4 categories as: stable, probation, flakey and lengthy/slow.

Reply:
Thanks @ghuangups for taking a look.

Initially I proposed to use text files listing tests for each category.  That's what we discussed offline.

In practice, however, I found it has a problem: using both include/exclude file and include/exclude list in POM (which we have for CI) is not really compatible.

Test list in text file also has two disadvantages compared to annotation-based tagging:
 * easier to make typo in test class/method name or pattern,
 * any renaming or other refactoring needs to be applied to the list as well, which is easy to forget.
Both of these are non-issues with tags.

Regarding the set of categories being defined:
 * Stable is everything that's not tagged, i.e. it is an implicit category.
 * Probation does not need to be tagged.  One can explicitly select any subset of the tests for execution via `-Dtest`, include file or Maven profile.  The CI-specific profiles are not active in this case.

Reply:
Thanks @adoroszlai for the great explanation. Yes, sure, regarding the current categories implementation. 
Regarding the tagging, how about using a small tool to do the tagging at run time based on the text file for test listing? My concern is the sustainability. For example, any test fix or moving between categories or the like will involve code level changes and go through whole CI process. On the other hand, if we have a text file based list, that could be a lot light-weighted?

Reply:
> any test fix or moving between categories or the like will involve code level changes and go through whole CI process

I hope that any test fix involves code change. ;)